e2e/ansible/playbooks/hyperconverged/test-replica-node-affinity/replica-node-affinity.yaml

```diff
@@ -0,0 +1,193 @@
+
+       - name: Replace the replica count in storage classes yaml
+         replace:
+           path: "{{ create_sc }}"
+           regexp: 'openebs.io/jiva-replica-count: "3"'
+           replace: 'openebs.io/jiva-replica-count: "2"'
+
+       - name: Delete the existing storage class and create new one
+         shell: kubectl delete sc openebs-percona; kubectl apply -f "{{ create_sc }}"
+         args:
+           executable: /bin/bash
+         register: sc_out
+         until: "'created' in sc_out.stdout"
+         delay: 10
+         retries: 5
+      
+       - name: Deploying Percona Application
+         shell: kubectl create -f "{{ percona_link }}"
+         args:
+           executable: /bin/bash
+         register: app_out
+         until: "'created' in app_out.stdout"
+         delay: 20
+         retries: 6
+       
+       - name: Confirm if the percona pod is running
+         shell: source ~/.profile; kubectl get pods --all-namespaces | grep percona
+         args:
+           executable: /bin/bash
+         delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+         register: result
+         until: "'percona' and 'Running' in result.stdout"
+         delay: 120
+         retries: 10
+   
+       - name: Check if the replica pods are created
+         shell: kubectl get pods --all-namespaces | grep rep | grep -i running |wc -l
+         args:
+           executable: /bin/bash
+         register: rep_count
+         until: "'2' in rep_count.stdout"
+         delay: 60
+         retries: 5
+         ignore_errors: true
+
+    # Getting PV name
+       - name: Obtain PV name
+         shell: kubectl get pv | grep openebs-percona | awk '{print $1}'
+         args:
+           executable: /bin/bash
+         register: pv_name
+
+       - name: Copy replica_patch.yaml to master
+         copy:
+           src: "{{ patch_file }}"
+           dest: "{{ result_kube_home.stdout }}"
+ 
+       - name: Get the first node name where replica1 is scheduled
+         shell: kubectl get po --all-namespaces -o wide | grep "{{ pv_name.stdout }}" | grep rep | awk '{print $8}' | awk 'FNR == 1 {print}'
+         args:
+           executable: /bin/bash
+         register: node1
+     
+       - name: Get the second node name where replica1 is scheduled
+         shell: kubectl get po --all-namespaces -o wide | grep "{{ pv_name.stdout }}" | grep rep | awk '{print $8}' | awk 'FNR == 2 {print}'
+         args:
+           executable: /bin/bash
+         register: node2
+    
+       - name: Update first node name in replica_patch file
+         replace:
+           path: "{{ patch_file }}"
+           regexp: 'nodename_where_replica_pod_1_got_scheduled'
+           replace: '{{ node1.stdout }}'
+     
+       - name: Update second node name in replica_patch file
+         replace:
+           path: "{{ patch_file }}"
+           regexp: 'nodename_where_replica_pod_2_got_scheduled'
+           replace: '{{ node2.stdout }}'
```

Comment:
Can the the previous steps of a) getting the individual node names b) replacing each node in patch file - be relooked/combined.